validator/client/src/main/java/tech/pegasys/teku/validator/client/loader/ValidatorSourceFactory.java

```diff
@@ -0,0 +1,155 @@
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.validator.client.loader;
+
+import java.io.File;
+import java.net.http.HttpClient;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import tech.pegasys.teku.core.signatures.SlashingProtector;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.service.serviceutils.layout.DataDirLayout;
+import tech.pegasys.teku.spec.Spec;
+import tech.pegasys.teku.validator.api.InteropConfig;
+import tech.pegasys.teku.validator.api.KeyStoreFilesLocator;
+import tech.pegasys.teku.validator.api.ValidatorConfig;
+import tech.pegasys.teku.validator.client.ValidatorClientService;
+
+public class ValidatorSourceFactory {
+  private final Spec spec;
+  private final ValidatorConfig config;
+  private final InteropConfig interopConfig;
+  private final Supplier<HttpClient> externalSignerHttpClientFactory;
+  private final SlashingProtector slashingProtector;
+  private final PublicKeyLoader publicKeyLoader;
+  private final AsyncRunner asyncRunner;
+  private final MetricsSystem metricsSystem;
+  private final Optional<DataDirLayout> maybeDataDir;
+  private Optional<ValidatorSource> mutableValidatorSource = Optional.empty();
+
+  public ValidatorSourceFactory(
+      final Spec spec,
+      final ValidatorConfig config,
+      final InteropConfig interopConfig,
+      final Supplier<HttpClient> externalSignerHttpClientFactory,
+      final SlashingProtector slashingProtector,
+      final PublicKeyLoader publicKeyLoader,
+      final AsyncRunner asyncRunner,
+      final MetricsSystem metricsSystem,
+      final Optional<DataDirLayout> maybeDataDir) {
+    this.spec = spec;
+    this.config = config;
+    this.interopConfig = interopConfig;
+    this.externalSignerHttpClientFactory = externalSignerHttpClientFactory;
+    this.slashingProtector = slashingProtector;
+    this.publicKeyLoader = publicKeyLoader;
+    this.asyncRunner = asyncRunner;
+    this.metricsSystem = metricsSystem;
+    this.maybeDataDir = maybeDataDir;
+  }
+
+  public List<ValidatorSource> createValidatorSources() {
+    final List<ValidatorSource> validatorSources = new ArrayList<>();
+    if (interopConfig.isInteropEnabled()) {
+      validatorSources.add(
+          slashingProtected(new MockStartValidatorSource(spec, interopConfig, asyncRunner)));
+    } else {
+      addExternalValidatorSource().ifPresent(validatorSources::add);
+      addLocalValidatorSource().ifPresent(validatorSources::add);
+      addMutableValidatorSource().ifPresent(validatorSources::add);
+    }
+    return validatorSources;
+  }
+
+  public Optional<ValidatorSource> getMutableValidatorSource() {
+    return mutableValidatorSource;
+  }
```

Comment:
We probably want to rename these to `mutableLocalValidatorSource` to be consistent with the class name.